File: LeetCode/TwoPointers/392. Is Subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if (s.isEmpty()) {
            return true;
        }
        if (s.length() > t.length()){
            return false;
        }
        int count = s.length();
        int index = 0;
        char p1 = s.charAt(0);
        for (int i = 0; i < t.length(); i++) {
            if (p1 == t.charAt(i)) {
                index++;
                if (index == count){
                    return true;
                }
                p1 = s.charAt(index);
            }
        }
        return false;
    }
}

File: LeetCode/TwoPointers/392. Is Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) == 0:
            return True
        current = 0
        lenght = len(s)
        for ch in t:
            if s[current] == ch:
                current += 1
            if current == lenght:
                return True
        return lenght == current
